### PR TITLE
Also add new_with_buffer_capacity

### PR DESCRIPTION
**Pull request by ryoqun**
we need to adjust the internal `BufWriter` capacity via its construction for best performance of super write intensive tasks: https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.with_capacity

so, add another method, not to break compatibility.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -166,13 +167,30 @@ where
...
     pub fn new<P>(path: P, condition: RC, max_files: usize) -> io::Result<RollingFileAppender<RC>>
+    where
+        P: AsRef<Path>,
+    {
+        Self::_new(path, condition, max_files, None)
+    }
+
+    /// Creates a new rolling file appender with the given condition and write buffer capacity.
+    /// The parent directory of the base path must already exist.
+    pub fn new_with_buffer_capacity<P>(path: P, condition: RC, max_files: usize, buffer_capacity: usize) -> io::Result<RollingFileAppender<RC>>
```
Comment (ryoqun) [inline src/lib.rs]: I considered `Appender::with_capacity`, straightly mimicing `BufWriter`'s counterpart name. but that would confuse from the view point of `rolling-file` crate users.

also, argument ordering can be contested as well. :)

**Comment by kevinhoffman** [comment]
@ryoqun - thanks for the PR, looks good. Could you run `cargo fmt` and resubmit so it will pass the test suite? Also, perhaps modify the tests `build_context` [function](https://github.com/Axcient/rolling-file-rs/blob/main/src/lib.rs#L307) to accept a new Option(usize) capacity arg and call new or new_with_capacity as appropriate, and then add a new test to test that when a large capacity is specified things work as designed?

**Comment by kevinhoffman** [comment]
@ryoqun - it also looks like clippy is failing now since `chrono::TimeZone::ymd` is deprecated it wants us to use `with_ymd_and_hms` -- chrono dependency could be explicitly advanced to require 0.4.23 and use the new function, or locked to 0.4.22 to avoid the deprecated function (might cause issues later). I'd be happy to address this first, or if you wanted to address that change with another commit in this PR too that is fine.

**Comment by ryoqun** [comment]
@kevinhoffman thanks for reply!

> Could you run cargo fmt and resubmit so it will pass the test suite?

done

> Also, perhaps modify the tests build_context [function](https://github.com/Axcient/rolling-file-rs/blob/main/src/lib.rs#L307) to accept a new Option(usize) capacity arg and call new or new_with_capacity as appropriate, and then add a new test to test that when a large capacity is specified things work as designed?

yeah done (mostly at https://github.com/Axcient/rolling-file-rs/pull/4/commits/33bb6c634d70b18349ad090d4cef6df5d833a985)

> it also looks like clippy is failing now since chrono::TimeZone::ymd is deprecated it wants us to use with_ymd_and_hms -- chrono dependency could be explicitly advanced to require 0.4.23 and use the new function, or locked to 0.4.22 to avoid the deprecated function (might cause issues later). I'd be happy to address this first, or if you wanted to address that change with another commit in this PR too that is fine.

after some research and thought, i decided to bump `chrono` version. it seems that this is what other crates are doing according to https://crates.io/crates/chrono/reverse_dependencies.

**Comment by kevinhoffman** [comment]
@ryoqun - wonderful. and thank you for fixing the clippy/chrono items too.

**Pull request merged**